[WPE] Launching MiniBrowser on DRM mode doesn't pick VSync clock

dpino opened the pull request:
#### 378c859a6e51b19ea340d33aa44de3e1c7e84c71
<pre>
[WPE] Launching MiniBrowser on DRM mode doesn&apos;t pick VSync clock
<a href="https://bugs.webkit.org/show_bug.cgi?id=297892">https://bugs.webkit.org/show_bug.cgi?id=297892</a>

Reviewed by Nikolas Zimmermann.

Function &apos;drmWaitVBlank&apos; needs a file descriptor with a frame buffer
attached in order to work. However, a trace shows the function call
that tries to attach the frame buffer, &apos;drmAddFrameBuffer&apos;, happens after.

To make &apos;drmWaitVBlank&apos; work before the frame buffer is ready, a mock
frame buffer is attached temporarily to the file descriptor before the
actual frame buffer is created.

Patch by Carlos Garcia Campos &lt;cgarcia@igalia.com&gt;

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.cpp:
(WPE::DRM::Crtc::Crtc):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.h:
(WPE::DRM::Crtc::bufferID const):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMSetup):
* Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRM.cpp:
(wpeScreenDRMCreateDumbBufferIfNeeded):
(wpeScreenDRMDestroyDumbBufferIfNeeded):
* Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRMPrivate.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(wpeViewDRMCommitAtomic):
(wpeViewDRMCommitLegacy):

Canonical link: <a href="https://commits.webkit.org/299155@main">https://commits.webkit.org/299155@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa5418fbb82f554e5583b5f30155ae5a02544e8b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118056 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37734 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28370 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124208 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70092 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38427 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46316 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89588 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59190 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121008 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30610 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105859 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70081 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29678 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67873 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100033 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24154 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127287 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44959 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33880 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98262 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45320 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102083 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98048 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43449 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21438 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41398 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18814 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44831 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50505 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44291 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47636 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45980 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRM.cpp:
```diff
@@ -41,6 +42,10 @@ struct _WPEScreenDRMPrivate {
+    struct {
+        uint32_t bufferID;
+        uint32_t frameBufferID;
+    } dumb;
```
:upside_down_face: the case where not knowing helps :sweat_smile:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fa5418fbb82f554e5583b5f30155ae5a02544e8b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49876

dpino committed:
[WPE] Launching MiniBrowser on DRM mode doesn't pick VSync clock
https://bugs.webkit.org/show_bug.cgi?id=297892

Reviewed by Nikolas Zimmermann.

Function 'drmWaitVBlank' needs a file descriptor with a frame buffer
attached in order to work. However, a trace shows the function call
that tries to attach the frame buffer, 'drmAddFrameBuffer', happens after.

To make 'drmWaitVBlank' work before the frame buffer is ready, a mock
frame buffer is attached temporarily to the file descriptor before the
actual frame buffer is created.

Patch by Carlos Garcia Campos <cgarcia@igalia.com>

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.cpp:
(WPE::DRM::Crtc::Crtc):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.h:
(WPE::DRM::Crtc::bufferID const):
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMSetup):
* Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRM.cpp:
(wpeScreenDRMCreateDumbBufferIfNeeded):
(wpeScreenDRMDestroyDumbBufferIfNeeded):
* Source/WebKit/WPEPlatform/wpe/drm/WPEScreenDRMPrivate.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(wpeViewDRMCommitAtomic):
(wpeViewDRMCommitLegacy):

Canonical link: https://commits.webkit.org/299155@main
